### PR TITLE
CON-2902-Markdown-Top-Frequents-Exercise

### PR DESCRIPTION
**Pull request by zanninso**
[CON-2902](https://01talent.atlassian.net/browse/CON-2902)

[CON-2902]: https://01talent.atlassian.net/browse/CON-2902?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: subjects/java/checkpoints/top-frequents/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+## Top Frequents
+
+### Instructions
+
+Given a non-empty array of integers, return the k most frequent elements. Use a hash map and a heap for efficient lookup and retrieval.
```
Comment (nprimo) [inline subjects/java/checkpoints/top-frequents/README.md:5]: I would add what is the sorting logic for the returned array: is it supposed to be ordered by "appearance" (smallest index in "original" array = smallest index in the returned array) or should it be based on "standard" ascending sorting of integer?

From the examples, both logics could be valid.

Comment (nprimo) [inline subjects/java/checkpoints/top-frequents/README.md:5]: What if k is greater than the number of unique elements in the array? Maybe adding an example to show what's expected could be enough

**Review by nprimo — CHANGES_REQUESTED** [review]

File: subjects/java/checkpoints/top-frequents/README.md (hunk 1 above)
Comment (zanninso) [reply subjects/java/checkpoints/top-frequents/README.md:5]: it should be ordred by appearance first then if there is there is frequent equality the one that have the smallest index in the original array will be first.

Comment (zanninso) [reply subjects/java/checkpoints/top-frequents/README.md:5]: okay i will add a test for it

**Review by nprimo — APPROVED** [review]
LGTM :rocket:

**Pull request merged**